File: strings/804. Unique Morse Code Words/uniqueMorseRepresentations.js
/**
 * @param {string[]} words
 * @return {number}
 */
var uniqueMorseRepresentations = function(words) {
    const morse = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."];
    let transformations = new Set();
    
    for (let word of words) {
        let transf = "";
        for (let char of word) {
            transf += morse[char.charCodeAt(0) - 97];
        }
        transformations.add(transf);
    }
    
    return transformations.size;
};

File: strings/804. Unique Morse Code Words/uniqueMorseRepresentations.py
class Solution(object):
    def uniqueMorseRepresentations(self, words):
        """
        :type words: List[str]
        :rtype: int
        """
        # Morse mapping for alphabet letters a-z in lower case
        morse = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
        # To store unique transformations
        transformations = set()
        
        for word in words:
            transf = ""
            for char in word:
                # ord(char): gets the ASCII, -97 to make it 0 indexed so we can map it to morse
                transf += morse[ord(char) - 97]
            transformations.add(transf)
        
        return len(transformations)